### ISSUE TITLE
peek and go to definiton cannot work after ctrl+v

### ISSUE THREAD

**Issue by Zhangbixu** [microsoft/vscode-cpptools#1662]
peek and go to definiton cannot work after ctrl+v until I restart VSCODE. When I hang the mouse on a variable, it displays "loading"; When I press F12, it cannot go to the definition.

**Comment by sean-mcmanus** [comment]
Is there a red flame in the bottom right? Can you wait for that to go away?

**Comment by Zhangbixu** [comment]
No, there is not any flame found there.

**Comment by sean-mcmanus** [comment]
Hmm...then we need more repro details to figure out what is causing this. We're not able to repro the problem and this is the first user report we've had of pasting causing our extension to be unusable. What OS and version of the extension are you using? What is the behavior of the Microsoft.VSCode.Extension.* process when it says loading? Are you able to get a call stack or dmp when it's in this state? Does the bug repro with a simple workspace with just a single C++ file with an empty main function? Does it repro when intelliSenseEngine is set to "Tag Parser"?

**Comment by Zhangbixu** [comment]
OS:WIN7,64bit
VSCODE:1.22.0insider
C/C++:0.16.0-insider
This bug did not appear in earlier C/C++ version,such as 0.13.0 or 0.14.0.
I tested it more and got some possibilities:
It is related with the total line numbers,if the last line of the file exceeds 200 or 300,then perform a paste, the bug appears. When it appears, the peek,go to definition, suggestions are disabled all.
Setting the intellisense engine to tag parser takes no effect either.

**Comment by sean-mcmanus** [comment]
Does it repro with 0.15.0? It tested it with a 202k line file and I'm not reproing the bug. Is there some randomness or timing involved? @bobbrow Any ideas how to repro this?

![image](https://user-images.githubusercontent.com/19859882/37318733-635ea9e2-2629-11e8-861b-ea989438e606.png)

**Comment by Zhangbixu** [comment]
I cannot roll back to 0.15.0, but I remember it was not OK either.
It is not random in my conputer, I repro this by ctrl+v repeatedly and it will repro soon.
Here are my installed extensitions.

Extension | Author (truncated) | Version
-- | -- | --
code-gnu-global | aus | 0.2.2
GBKtoUTF8 | buk | 0.0.2
dakara-foldplus | dak | 1.2.2
vscode-eslint | dba | 1.4.7
changeEncode | lan | 0.0.8
vscode-clang | mit | 0.2.2
cpptools | ms- | 0.16.0-insiders
sublime-keybindings | ms- | 3.0.3


Here is my test file, you can repro this by copy the last few lines and paste them into the end, maybe it takes several times to repro this.
[test2.zip](https://github.com/Microsoft/vscode-cpptools/files/1809826/test2.zip)

**Comment by sean-mcmanus** [comment]
It says "test2.zip couldn't be downloaded". Can you put it somewhere else?

It should be possible to uninstall 0.16.0 and then install the published 0.15.0.

**Comment by Zhangbixu** [comment]
I tested 0.15.0 again and it repro too.
I put the test file here, you can copy it to a file named helloworld.c, put it in a directory and open it with vscode. Then you can repro this by copy the last few lines and paste them into the end, maybe it takes several times to repro this. I repro this for only once.

#include <stdio.h>
void foo(void)
{

}
void main(void)
{
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    printf("hello world!\n");
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    foo();
    printf("hello world!\n");
    printf("hello world!\n");
}

**Comment by sean-mcmanus** [comment]
I'm still not able to repro a problem. What lines exactly are you pasting? And where? I tried pasting The foo/printf lines inside the main method (i.e. so the code is syntactically correct), and I got to 5k lines of pasting and Go to Definition still works.

**Comment by Zhangbixu** [comment]
I paste only one printf line inside the main method and it repros everytime.
I tested vscode 1.14.2 and 1.17 and it has the same problem, so I think it is not related with the lastest vscode version.

**Comment by sean-mcmanus** [comment]
I'm still not able to repro the issue. What encoding is the file? UTF-8? Could another extension be causing a problem?

**Comment by Zhangbixu** [comment]
The encoding is UTF-8. I uninstalled other extensions and it is still not OK.
I tested in another computer, which is win10 64bit, and it does not repro. So maybe it is related with the OS version.But about half a year ago(201707), my computer,which is win7 64bit,was able to go well with the vscode and c/cpp extension.

**Comment by sean-mcmanus** [comment]
I tried Windows 7, but I'm still not able to repro any failures after pasting. Can you set the loggingLevel to "6" and look for errors? It sounds like your symbol database is becoming corrupt, although we don't update the database until a save occurs, and it sounds like you're hitting the error without saving?

**Comment by sean-mcmanus** [comment]
0.16.1 fixes a bug that could randomly lead to a deadlock and endless "Loading..." ( #1759 ). I'm not sure if your issue had the same root cause or not.

**Comment by bobbrow** [comment]
I'm closing old issues marked "more info needed".  If you are still having issues with the extension, please ensure you have installed the latest version and open a new issue with the issue details.
